src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -890,14 +852,13 @@ LookupCitusTableCacheEntry(Oid relationId)
 	HOLD_INTERRUPTS();
 
 	/* actually fill out entry */
-	BuildCitusTableCacheEntry(cacheEntry);
+	CitusTableCacheEntry *cacheEntry = InitCitusTableCacheEntry(relationId);
 
-	/* and finally mark as valid */
-	cacheEntry->isValid = true;
+	cacheSlot->ref = InitCitusTableCacheEntryRef(MetadataCacheMemoryContext, cacheEntry);
```

Comment:
Since refs can't be pfree'd, it may be better to not leak these small objects in `MetadataCacheMemoryContext`. Instead creating light weight references which must be manually released *(read: total head ache alongside errors)* which could then allow the cache slot to only be a pointer to the `CitusTableCacheEntry`